src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/KeyValuePairConverter.cs

```diff
@@ -70,28 +104,21 @@ internal sealed class KeyValuePairConverter<TKey, TValue> : JsonValueConverter<K
             }
 
             propertyName = reader.GetString()!;
-            if (propertyName == KeyName)
+            if (!keySet && FoundKeyProperty(propertyName, caseInsensitiveMatch))
```

Comment:
So this would break someone if all of these conditions are true:
- Case insensitivity off
- Previously serialized JSON is deserialized ("Key" and "Value")
- There is a naming policy such as camel-casing that changes "Key"\"Value" to something else.

Reply:
That was what we last discussed (offline). However, by adding specific checks for ["Key"](https://github.com/dotnet/runtime/pull/36424/files#diff-17f25b109d239741ebe3cc5c877d272dR151) and ["Value"](https://github.com/dotnet/runtime/pull/36424/files#diff-17f25b109d239741ebe3cc5c877d272dR158), we end up not breaking any one wrt. deserialization.

Reply:
I see. I missed the check for explicit clr name.